platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationEntry.java

```diff
@@ -0,0 +1,108 @@
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+
+/**
+ * The <code>MigrationEntry</code> interface provides support for artifacts that are being exported
+ * or imported during migration.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface MigrationEntry extends Comparable<MigrationEntry> {
+    /**
+     * Gets the migration report associated with this entry. Warnings or errors can be recorded with
+     * the report.
+     *
+     * @return the migration report associated with this entry
+     */
+    public MigrationReport getReport();
+
+    /**
+     * Gets the identifier for the {@link Migratable} service responsible for this entry.
+     *
+     * @return the responsible migratable service id
+     */
+    public String getId();
+
+    /**
+     * Gets name for this entry. The name is standardized with slashes (i.e /).
+     *
+     * @return the name for this entry
+     */
+    public String getName();
+
+    /**
+     * Gets a {@link Path} for this entry.
+     * <p>
+     * <i>Note:</i> Absolute paths that are under ${ddf.home} are automatically relativized.
+     *
+     * @return a path for this entry
+     */
+    public Path getPath();
+
+    /**
+     * Gets the last modification time of the entry.
+     *
+     * @return the last modification time of the entry in milliseconds since the epoch,
+     * or -1 if not specified
+     */
+    public long getLastModifiedTime();
+
+    /**
+     * Stores this entry's content in the export during an export migration operation or underneath
+     * the distribution root directory during an import migration operation based on this entry's
+     * path which can include sub-directories.
+     * <p>
+     * During an import migration operation, this entry's sub-directories (if any) will be created if
+     * they don't already exist. The destination file will be overwritten if it already exists.
+     * <p>
+     * All errors and warnings are automatically recorded with the associated migration report.
+     *
+     * @return <code>true</code> if no errors were recorded as a result of processing this command;
+     * <code>false</code> otherwise
+     * @throws MigrationException if a failure that prevents the operation from continuing occurred
+     */
+    public default boolean store() {
```

Comment:
❗️ Putting the `store` methods here forces both, `ImportMigrationEntry` and `ExportMigrationEntry` to share the same method names, which doesn't make much sense on the import side. Could we either move those up to those two interfaces and use more appropriate names (e.g., `store` and `restore`) or, if we keep them here, use more generic names like `process`?

Reply:
The initial design had store completely defined in the base class and then the overloaded versions came to be and required defining them in the subclasses.
process() was my second choice but I do love store() and restore(). I am adding it to the todo list for the next PR.